From 014be9171d472255dac5f6e8fba21514261b2237 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Fri, 8 Jul 2016 15:23:25 -0500
Subject: [PATCH 14/21] Changing the way miscellaneous output register is set
 for IGA1

Now only certain bits of miscellaneous output register meant for IGA1
are set within viaIGA1SetDisplayRegister function.

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/src/via_display.c b/src/via_display.c
index 7738516..4e28172 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -885,15 +885,24 @@ viaIGA1SetDisplayRegister(ScrnInfoPtr pScrn, DisplayModePtr mode)
     xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                 "IGA1 Requested Screen Mode: %s\n", mode->name);
 
-    /* Set Misc Register */
-    temp = 0x23;
-    if (mode->Flags & V_NHSYNC)
+    /* Set certain bits of miscellaneous output register
+     * meant for IGA1. */
+    temp = hwp->readMiscOut(hwp);
+    if (mode->Flags & V_NHSYNC) {
         temp |= 0x40;
-    if (mode->Flags & V_NVSYNC)
+    } else {
+        temp &= (~0x40);
+    }
+
+    if (mode->Flags & V_NVSYNC) {
         temp |= 0x80;
-    temp |= 0x0C; /* Undefined/external clock */
+    } else {
+        temp &= (~0x80);
+    }
+
     hwp->writeMiscOut(hwp, temp);
 
+
     /* Sequence registers */
     hwp->writeSeq(hwp, 0x00, 0x00);
 
-- 
1.9.1

